yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -1414,12 +1414,15 @@ func getAssessmentIssuesForUnsupportedDatatypes(unsupportedDatatypes []utils.Tab
 			}
 		case POSTGRESQL:
 			// Datatypes can be of form public.geometry, so we need to extract the datatype from it
-			var datatype string
-			if strings.Contains(colInfo.DataType, ".") {
-				datatype = strings.Split(colInfo.DataType, ".")[1]
-			} else {
-				datatype = colInfo.DataType
+			datatype, ok := utils.SliceLastElement(strings.Split(colInfo.DataType, "."))
+			if !ok {
+				log.Warnf("failed to get datatype from %s", colInfo.DataType)
+				continue
 			}
+
+			// We obtain the queryissue from the Report function. This queryissue is first converted to AnalyzeIssue and then to AssessmentIssue using pre existing function
+			// Coneverting queryissue directly to AssessmentIssue would have lead to the creation of a new function which would have required a lot of cases to be handled and led to code duplication
+			// This converted AssessmentIssue is then appended to the assessmentIssues slice
```

Comment:
I am just wondering that for analyze schema issues we detect the unsupported datatypes based on scanning the DDL in schema files.

Here we are detecting based on system catalog containing each tables' column dataypes.

Two things I am not sure(not necessarily because of this PR):
1. How is it ensured that those issues are not reported twice.
2. Can we make use of same logic in both - assess and analyze, maybe just use the issues returned from analyze schema?

Reply:
Dual reporting won't happen because in assess we fetch each analyze issue specifically by passing the Issue Type info.